python/cudf/cudf/core/dataframe.py

```diff
@@ -12,6 +12,7 @@
 import textwrap
 import warnings
 from collections import abc, defaultdict
+from types import NotImplementedType
```

Comment:
I'm fine with this change, but I'm curious: do newer versions of mypy now complain about this?

Reply:
Ah here's the issue. `NotImplementedType` was only [introduced in Python 3.10](https://docs.python.org/3/library/types.html#types.NotImplementedType), so we can't use that here.

Reply:
Ah, let me check in 3.8 and 3.9

Reply:
Added a workaround

Reply:
> I'm fine with this change, but I'm curious: do newer versions of mypy now complain about this?

Yes: https://github.com/python/mypy/issues/4791

I've backwards-compated by setting `NotImplementedType = Any` because I can't see another way of doing things.

Reply:
Yeah, we need a `sys.version` gated import which is now done.